client/src/pages/Register.js

```diff
@@ -59,43 +59,53 @@ export default function Register() {
   const [name, setName] = useState('');
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
   const [primaryLanguage, setPrimaryLanguage] = useState('');
 
   const [errorName, setErrorName] = useState(false);
   const [errorEmail, setErrorEmail] = useState(false);
   const [errorPassword, setErrorPassword] = useState(false);
+  const [errorConfirmPassword, setErrorConfirmPassword] = useState(false);
   const [errorLanguage, setErrorLanguage] = useState(false);
 
   const isEmail = (email) => /^\S+@\S+$/.test(email);
-  const isName = (name) => /^[A-Z]+$/i.test(name);
+  //const isName = (name) => /^[A-Z]+$/i.test(name);
 
-  const errors = () => {
-    return (
-      errorName ||
-      errorEmail ||
-      errorPassword ||
-      errorLanguage ||
-      !userState.user.errorMsg === ''
-    );
+  const isErrorFree = () => {
+    //if (!isName(name)) return false;
+    if (!name) return false;
+    if (!isEmail(email)) return false;
+    if (password.length <= 5) return false;
+    if (!primaryLanguage) return false;
+    if (password !== confirmPassword) return false;
+    if (!userState.user.errorMsg === '') return false;
+
+    return true;
   };
 
-  const handleSubmit = () => {
+  const handleSubmit = async () => {
     if (!isEmail(email)) setErrorEmail(true);
-    if (!isName(name)) setErrorName(true);
+    //if (!isName(name)) setErrorName(true);
+    if (!name) setErrorName(true);
     if (!password) setErrorPassword(true);
     if (!primaryLanguage) setErrorLanguage(true);
+    if (password !== confirmPassword) {
+      setErrorConfirmPassword(true);
+    }
```

Comment:
Can we move all this into isErrorFree? We only want to check each thing once.
Also, we don't need it async unless we plan on awaiting something